Dev 488 change donation to contribution

x110dc opened the pull request:
#### Plan? (if this draft/incomplete indicate what you intend to do and how)

To rename user-visible mentions of "Donations" to "Contributions"

#### What's this PR do?

- Replaces "Donations" on the Org Portal tab with "Contributions"
- Replaces mentions of Donations with Contributions in the README

#### Why are we doing this? How does it help us?

- Contributions is broader than donations. This will make our personas and other terms more consistent.

#### How should this be manually tested?

- Look at the Org Admin tabs and confirm you see "Contributions" and not "Donations"

#### Have automated unit tests been added?

No. Just copy changes.

#### How should this change be communicated to end users?

N/A

#### Are there any smells or added technical debt to note?

I saw at least one possible URL as `/donations/` that I didn't change because I wasn't sure of the implications or what the potentially broken link would mean. Opened as a separate ticket: [DEV-488]

#### Has this been documented? If so, where?

No need, I don't think.

#### What are the relevant tickets?
**(Note: Please use syntax [JIRA-123] to link any relevant tickets)**

[DEV-488]

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them?
**(Note: Please use syntax [JIRA-123] to link any relevant tickets)**

Yes: 
- https://news-revenue-hub.atlassian.net/browse/DEV-1446
- https://news-revenue-hub.atlassian.net/browse/DEV-1464 

[DEV-488]: https://news-revenue-hub.atlassian.net/browse/DEV-488?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

benjaminEwhite reviewed (commented):
Found additional references with search for "donation".

## In Django app:

- It looks like in apps/emails, there's still "Donation" language. I didn't look much into it beyond noting presence of keyword in file, but probably emails are referring to "donations", I would think.

## In SPA:

- In spa/src/components/donations/Donations.js at lines 134 and 139 there are headings with the word "donation".
- In spac/src/components/donations/DonationDetail.js 
  - at line 120, there's a heading attribute that needs to be updated
  - line 95, there's a reference to donation in user confirmation message text
 - Probably want to update spa/src/routes.js L4 for "DONATIONS_SLUG", as that determines the URL for contribution-related pages
 - spa/src/components/donationPage/pageContent/SPageHeading.js L20 - "This is a static element at the top of the donation section..."
 - spa/src/components/donationPage/pageContent/SHeaderBar L15
 - spa/src/components/donationPage/pageContent/DSwag.js L80
 - spa/src/components/donationPage/pageContent/DRichText.js L16
 - spa/src/components/donationPage/pageContent/DFrequency.js L63
 - spa/src/components/donationPage/pageContent/DAmount.js L118, L119
 - spa/src/components/connect/stripe/StripeProvider.js L89
 - spa/src/compoonents/content/pages/AddPageModal.js L135
 - spa/src/components/analytics/plugins/facebookPixel.js L58 (maybe? -- would require spa/src/analytics/Analytics.spec.js L38 to also change)
 - spa/src/components/donationPage/live/LivePage404.js L8 (maybe?)

Separate from this ticket/PR, might make sense to create a ticket to change references to Donation to Contribution in the SPA components (not just copy) — for instance, DonationPage would become ContributionPage.

x110dc commented:
Thanks Ben. 

- I didn't find any user-facing strings in the Django part in `apps/emails` but I did find reference to some templates stored in Mailgun. I copied those to the repo so we have a sort-of version control on them, updated them, and copied them back. 
- I updated most of the SPA references you mentioned. 
- I already had a ticket to update the models themselves (see the "next steps" section of the PR. That's also where I'd like to make the change to the URL. 

Re-requesting your review.

benjaminEwhite reviewed (approved):
LGTM
